test/config/test/workers-option.test.ts

```diff
@@ -0,0 +1,33 @@
+import { type UserConfig, describe, expect, test, vi } from 'vitest'
+
+import { getWorkersCountByPercentage } from 'vitest/src/utils/workers.js'
+import * as testUtils from '../../test-utils'
+
+function runVitest(config: UserConfig) {
+  return testUtils.runVitest({ ...config, root: './fixtures/workers-option' })
+}
+
+test('workers percent argument should not throw error', async () => {
+  const { stderr } = await runVitest({ maxWorkers: '100%', minWorkers: '10%' })
+
+  expect(stderr).toBe('')
+})
+
+describe('workers util', () => {
+  vi.mock('node:os', async importOriginal => ({
```

Comment:
`vi.mock `is always hoisted to the top of the file so it doesn't make a sense to have it inside `describe` function, let's put it at the top to make it less confusing

Reply:
Now that I look at it, I wonder how `vi.mock` worked. It's reflected.